slobrok/src/vespa/slobrok/server/map_view.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "map_view.h"
+#include <vespa/log/log.h>
+LOG_SETUP(".slobrok.map_view");
+
+namespace slobrok {
+
+MapView::MapView()
+  : _map(),
+    _currGen(0),
+    _lock()
+{}
+
+MapView::~MapView() {
+    clear();
+}
+
+void MapView::apply(const MapDiff &diff) {
+    std::lock_guard guard(_lock);
+    LOG(debug, "Applying diff from gen %u", diff.fromGen.getAsInt());
```

Comment:
consider adding assert that from gen matches (map vs diff) to indicate that this needs to be handled externally.